src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/DepartureTransformer.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.health.api.model.Departure
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DepartureEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DepartureReason
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.MoveOnCategory
+
+@Component
+class DepartureTransformer(private val destinationProviderTransformer: DestinationProviderTransformer) {
+  fun transformJpaToApi(jpa: DepartureEntity?): Departure? {
+    if (jpa == null) return null
+
+    return Departure(
+      bookingId = jpa.booking.id,
+      dateTime = jpa.dateTime,
+      reason = transformDepartureReason(jpa.reason),
+      moveOnCategory = transformMoveOnCategory(jpa.moveOnCategory),
+      destinationProvider = destinationProviderTransformer.transformJpaToApi(jpa.destinationProvider),
+      notes = jpa.notes
+    )
+  }
+
+  private fun transformDepartureReason(departureReason: DepartureReason): Departure.Reason {
+    return when (departureReason) {
+      DepartureReason.Absconded -> {
+        Departure.Reason.absconded
+      }
+      DepartureReason.PlannedMove -> {
+        Departure.Reason.plannedMove
+      }
+      DepartureReason.Recalled -> {
+        Departure.Reason.recalled
+      }
+    }
+  }
```

Comment:
Can you explain how this works? `DepartureReason` is an enum, and it looks like `Departure.Reason` is a subclass of `Departure`, but I don't see where this is.

Reply:
This is no longer in the code - but to answer the question - DepartureReason is an enum in the JPA package that I had written by hand wherreas Departure.Reason was in the generated API response models package that OpenAPI generate produces.  Departure.Reason also being an enum but as a nested class rather than at the package level.  e.g.

I'd written as:
```
package xyz.jpa

class DepartureEntity { ... }
enum class DepartureReason { enum1, enum2 .... }
```

OpenAPI had generated as:
```
package xyz.models
class Departure {
   ....
   enum class Reason { enum1, enum2 }
}
```

Reply:
Ah ha! That makes sense. Thanks for the explanation 👍